src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java

```diff
@@ -0,0 +1,280 @@
+package com.rackspace.ceres.app.services;
+
+
+import static com.rackspace.ceres.app.web.TagListConverter.convertPairsListToMap;
+
+import com.rackspace.ceres.app.config.AppProperties;
+import com.rackspace.ceres.app.config.DownsampleProperties;
+import com.rackspace.ceres.app.config.DownsampleProperties.Granularity;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.data.cassandra.core.cql.ReactiveCqlTemplate;
+import org.springframework.data.redis.core.ReactiveStringRedisTemplate;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Sinks.Empty;
+
+@Service
+@Slf4j
+@Profile("downsample")
+public class MetricDeletionService {
+
+  private static final String PREFIX_SERIES_SET_HASHES = "seriesSetHashes";
+  private static final String DELIM = "|";
+
+  private final ReactiveCqlTemplate cqlTemplate;
+  private final DataTablesStatements dataTablesStatements;
+  private AppProperties appProperties;
+  private TimeSlotPartitioner timeSlotPartitioner;
+  private DownsampleProperties downsampleProperties;
+  private final ReactiveStringRedisTemplate redisTemplate;
+
+  @Autowired
+  public MetricDeletionService(ReactiveCqlTemplate cqlTemplate,
+      DataTablesStatements dataTablesStatements, AppProperties appProperties,
+      TimeSlotPartitioner timeSlotPartitioner,
+      DownsampleProperties downsampleProperties, ReactiveStringRedisTemplate redisTemplate) {
+    this.cqlTemplate = cqlTemplate;
+    this.dataTablesStatements = dataTablesStatements;
+    this.appProperties = appProperties;
+    this.timeSlotPartitioner = timeSlotPartitioner;
+    this.downsampleProperties = downsampleProperties;
+    this.redisTemplate = redisTemplate;
+  }
+
+  public Mono<Empty> deleteMetrics(String tenant, String metricName, List<String> tag,
+      Instant start, Instant end) {
+    if (StringUtils.isAllBlank(metricName)) {
+      return deleteMetricsByTenantId(tenant, start, end);
+    } else if (CollectionUtils.isEmpty(tag)) {
+      return deleteMetricsByMetricName(tenant, metricName, start, end);
+    } else {
+      return deleteMetricsByMetricNameAndTag(tenant, metricName, tag, start, end);
+    }
+  }
+
+  private Mono<Empty> deleteMetricsByTenantId(String tenant, Instant start, Instant end) {
+    log.info("inside deleteMetricsByTenantId method with tenant {} ", tenant);
```

Comment:
These log messages need to be fixed. Maybe make it a debug level. If we want the information level logging we should place that in the controller.

"Deleting metrics for tenant: {}" would be be a reasonable message.